src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -601,6 +677,46 @@ public ProjectState AddAdditionalDocument(TextDocumentState document)
                 additionalDocumentStates: _additionalDocumentStates.Add(document.Id, document));
         }
 
+        public ProjectState AddAnalyzerConfigDocuments(ImmutableArray<AnalyzerConfigDocumentState> documents)
+        {
+            Debug.Assert(!documents.Any(d => this._analyzerConfigDocumentStates.ContainsKey(d.Id)));
+
+            var newAnalyzerConfigDocumentStates = _analyzerConfigDocumentStates.AddRange(documents.Select(d => KeyValuePairUtil.Create(d.Id, d)));
```

Comment:
i'm not really certain i undertand what you're doing here.

Reply:
_analyzerConfigDocumentStates is an ID to State map, so this is doing it that way.